docs/scripts/util/checkLink.ts

```diff
@@ -0,0 +1,81 @@
+import https from 'https';
+import http from 'http';
+import { IGNORED_LINKS } from '../../src/data/ignoredLinks';
+
+const requestedUrl: Set<string> = new Set();
+export async function checkLink(
+  {
+    href,
+    tagName,
+    tagText,
+    pageIdx,
+    pageUrl,
+  }: {
+    href: string;
+    tagName: string;
+    tagText: string;
+    pageIdx: string;
+    pageUrl: string;
+  },
+  linkIdx: number
+) {
+  return new Promise(async (res, rej) => {
+    if (!href) {
+      console.log(
+        `⚠️[WARNING...] page #${pageIdx} link #${linkIdx} "${tagName}" tag "${tagText}" doesn't have a href.`
+      );
+      res(0);
+    } else if (IGNORED_LINKS.includes(href) || requestedUrl.has(href)) {
+      console.log(
+        `⏭[SKIPPING...] page #${pageIdx} link #${linkIdx} ${href} from ${tagName} tag "${tagText}" on page ${pageUrl}, because it is on the IGNORED_LINKS list or have already requested.`
+      );
+      res(0);
+    } else {
+      const { get } = href.includes('https:') ? https : http;
+      const request = await get(href, async ({ statusCode = 0 }) => {
+        await returnStatus({ statusCode, href });
+        requestedUrl.add(href);
+        res(statusCode);
+      });
+      request.end();
+    }
+  });
+
+  async function returnStatus({
+    statusCode,
+    href,
+  }: {
+    statusCode: number;
+    href: string;
+  }) {
+    if ([200, 301, 303, 308].includes(statusCode)) {
+      console.log(
+        `↩️ [RETURNING STATUS...] ${statusCode} page #${pageIdx} link #${linkIdx} -- ${href} from ${tagName} tag "${tagText}" on page ${pageUrl}`
+      );
+
+      /**
+       * If 308, check if it's a internal direction (see docs/next.config.js redirects logic)
+       * If it's internal direction, after adding the platform, it should be 200
+       * Otherwise, the link needs to be updated
+       */
+      if (statusCode === 308) {
+        const hostNameRegex = /http(s)?:\/\/[^/]*/i;
```

Comment:
Curious if we can use a hardcoded string here rather than regex?

Reply:
yes, there's no difference from `const hostNameRegex = '/http(s)?:\/\/[^/]*/i';`

Reply:
Cool! Can we replace with a hard-coded string then?

Reply:
If no difference, why change?

Reply:
Because regex patterns are hard to grok, whereas a hard coded string is exactly as it reads

Reply:
+1, regex is really hard to read and prone to cause misunderstandings

Reply:
Done 8e3232b5f035923fefe5400a54c537ce125cf1fa

Reply:
Forgot to include proper test cases in the test 😂  String actually don't work, so I refactorec to `RegRex()` method and add a comment e7ecf9b5b9ae41828a4d0d4d9381977ab2ae574a